src/factory/draft/preset_discovery.rs

```diff
@@ -0,0 +1,228 @@
+pub struct clap_preset_discovery_metadata_receiver {
+    pub receiver_data: *mut c_void,
+    pub on_error: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            os_error: i32,
+            error_message: *const c_char,
+        ),
+    >,
+    pub begin_preset: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            name: *const c_char,
+            load_key: *const c_char,
+        ) -> bool,
+    >,
+    pub add_plugin_id: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            plugin_id: *const clap_plugin_id,
+        ),
+    >,
+    pub set_soundpack_id: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            soundpack_id: *const c_char,
+        ),
+    >,
+    pub set_flags: Option<
+        unsafe extern "C" fn(receiver: *const clap_preset_discovery_metadata_receiver, flags: u32),
+    >,
+    pub add_creator: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            creator: *const c_char,
+        ),
+    >,
+    pub set_description: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            description: *const c_char,
+        ),
+    >,
+    pub set_timestamps: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            creation_time: clap_timestamp,
+            modification_time: clap_timestamp,
+        ),
+    >,
+    pub add_feature: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            feature: *const c_char,
+        ),
+    >,
+    pub add_extra_info: Option<
+        unsafe extern "C" fn(
+            receiver: *const clap_preset_discovery_metadata_receiver,
+            key: *const c_char,
+            value: *const c_char,
+        ),
+    >,
+}
+
+unsafe impl Send for clap_preset_discovery_metadata_receiver {}
+unsafe impl Sync for clap_preset_discovery_metadata_receiver {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_preset_discovery_filetype {
+    pub name: *const c_char,
+    pub description: *const c_char,
+    pub file_extension: *const c_char,
+}
+
+unsafe impl Send for clap_preset_discovery_filetype {}
+unsafe impl Sync for clap_preset_discovery_filetype {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_preset_discovery_location {
+    pub flags: u32,
+    pub name: *const c_char,
+    pub uri: *const c_char,
+}
+
+unsafe impl Send for clap_preset_discovery_location {}
+unsafe impl Sync for clap_preset_discovery_location {}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct clap_preset_discovery_soundpack {
+    pub flags: u32,
```

Comment:
Should be `u64`: https://github.com/free-audio/clap/blob/065d685d4e9657f0344f350eef748be2b4d8e318/include/clap/factory/draft/preset-discovery.h#L203

Reply:
u32 is actually correct here. The spec is inconvenient and uses both interchangeably. But it's primarily used as u32.

Reply:
I don't understand; the struct layouts of `clap_preset_discovery_soundpack` will not match if the CLAP headers use `uint64_t` and `clap-sys` uses `u32`.

Reply:
Oops sorry, I opened this on my phone and I thought the comment was about the `u32` type for the `CLAP_PRESET_DISCOVERY_IS_*` flags. Fixed, along with the other errors.